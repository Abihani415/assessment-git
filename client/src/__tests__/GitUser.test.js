import React from "react";
import { screen, render, fireEvent, act, waitForElementToBeRemoved } from '../test-utils'
import "@testing-library/jest-dom";
import GitUser from "../component/gitUser";


describe("Take GitUser Component snapshot" , () => {
  it("User Detail", async () => {
    render(<GitUser />)
    const inputField = screen.getByTestId('username-test-id');
    const submitBtn = screen.getByTestId('submit-test-id');

    fireEvent.change(inputField, {target: {value: 'ankitbihani415'}});
    await act(async () => {
        fireEvent.click(submitBtn);
    });
    await waitForElementToBeRemoved(() => screen.queryAllByTestId("loading-test-id"))
    expect(screen.getByTestId("user-profile-test-id")).toMatchSnapshot();
  })
})