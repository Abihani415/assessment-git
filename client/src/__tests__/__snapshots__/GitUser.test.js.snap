// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Take GitUser Component snapshot User Detail 1`] = `
<div
  class="card"
  data-testid="user-profile-test-id"
>
  <div
    class="d-flex align-items-center p-3 flex-column"
  >
    <a
      href="https://github.com/ankitbihani415"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="ankitbihani415"
        class="card-img avatar"
        src="https://avatars.githubusercontent.com/u/66110273?v=4"
      />
    </a>
    <div
      class="mb-4 mt-3 card-title h5"
    >
      ankitbihani415
      <div
        class="date"
      >
        29/5/2020
      </div>
    </div>
  </div>
  <div
    class="border-top p-2 card-body"
  >
    <div
      class="d-flex align-items-center justify-content-between"
    >
      <div>
        <span>
          Follower: 
          0
        </span>
      </div>
      <div>
        <span>
          Following: 
          0
        </span>
      </div>
      <div>
        <span>
          Total Repo: 
          7
        </span>
      </div>
    </div>
  </div>
</div>
`;
